Frontend/src/pages/register/components/RegistrationPanel.jsx

```diff
@@ -3,27 +3,22 @@ import React, { useContext, useState } from 'react'
 import { useParams } from 'react-router-dom'
 import { Button, TextArea } from 'semantic-ui-react'
 import { AuthContext } from '../../../api/helper/context/auth_context'
-import { useHeldEvents } from '../../../api/helper/hooks/use_competition_info'
 import submitEventRegistration from '../../../api/registration/post/submit_registration'
 import { setMessage } from '../../../ui/events/messages'
-import LoadingMessage from '../../../ui/messages/loadingMessage'
 import styles from './panel.module.scss'
+import { CompetitionContext } from '../../../api/helper/context/competition_context'
 
 export default function RegistrationPanel() {
   const [selectedEvents, setSelectedEvents] = useState([])
   const { competition_id } = useParams()
   const { user } = useContext(AuthContext)
-  const { isLoading, heldEvents } = useHeldEvents(competition_id)
+  const { competitionInfo } = useContext(CompetitionContext)
   const [comment, setComment] = useState('')
```

Comment:
Generally, it would be nice to have the 'global' things - contexts and params - at the top, and 'local' things - states - at the bottom. And a new line before the return.
Same comment applies to a few other components.

Reply:
where does useQuery come in the order? I think otherwise it should be good

Reply:
Uh maybe after all the things I already listed?
Obviously the order isn't super important, but I mentioned it because seeing `selectedEvents` and `comment` so far apart felt wrong.